plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java

```diff
@@ -88,9 +89,11 @@
 
   private Text version;
 
-  private Button autoPromoteButton;
+  @VisibleForTesting
+  Button autoPromoteButton;
```

Comment:
There should be a way to locate the button in the widget by starting at the parent that does not require access to the private fields. Just walk down the widget tree looking for a Button with the right name/ID. Not sure it's worth it though. Doesn't look like SWT supports this as well as it should.

http://stackoverflow.com/questions/17945611/how-to-assign-unique-id-to-swt-widget-and-get-it-in-swtbot